lib/auth/auth_test.go

```diff
@@ -358,6 +358,20 @@ func (s *AuthSuite) TestBadTokens(c *C) {
 	c.Assert(err, NotNil)
 }
 
+func (s *AuthSuite) TestGenerateTokenEventsEmitted(c *C) {
+	eventEmitted := false
+	s.mockedAuditLog.MockEmitAuditEvent = func(event events.Event, fields events.EventFields) error {
+		eventEmitted = true
+		c.Assert(event, DeepEquals, events.TrustedClusterTokenCreate)
+		return nil
+	}
+
+	// test trusted cluster token
+	_, err := s.a.GenerateToken(GenerateTokenRequest{Roles: teleport.Roles{teleport.RoleTrustedCluster}})
+	c.Assert(err, IsNil)
+	c.Assert(eventEmitted, Equals, true)
+}
```

Comment:
add a case for multiple roles too

Reply:
revised: https://github.com/gravitational/teleport/pull/3733/commits/e080552a14dc6e0518640108b507b544012a7d19